core/api/usermanager/v1/usermanager_proto/usermanager.proto

```diff
@@ -0,0 +1,156 @@
+  // add_signing_keys specifies whether to add this service's signing keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed.
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_signing_keys = 7;
+}
+
+// UpdateUserRequest sets the data field for the user.
+// The user must have the service's current signing key in its list of
+// authorized_keys in order to succeed.
+message UpdateUserRequest {
+  // user contains data which will be applied to the user.
+  keytransparency.v1.proto.User user = 2;
+  // update_mask specifies which fields of user to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 3;
+}
+
+// BatchCreateUserRequest creates multiple users all at once.
+message BatchCreateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // users is the set of users to create.
+  repeated keytransparency.v1.proto.User users = 3;
+  // add_signing_keys specifies whether to add this service's signing_keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed.
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_signing_keys = 4;
+}
+
+// BatchCreateUserRequest creates multiple users at once.
+message BatchCreateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+// BatchUpdateUserRequest creates multiple users all at once.
+message BatchUpdateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // users is the set of users to create.
+  repeated keytransparency.v1.proto.User users = 3;
+  // update_mask specifies which fields of user to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 4;
+}
+
+// BatchUpdateUserRequest creates multiple users at once.
+message BatchUpdateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+
+// The UserManagerService holds a set of signing keys which can be used to
+// sign user updates.
+// - This service must be registered with the server in order to submit user
+// create or update requests.
+// - The update requests this service generates will only be valid for users
+// that have added this service's signing keys to their list of authorized_keys.
+// - This service should carefully controll and log access to it's APIs.
+// - This service should rotate its signing keys periodically.
+//   New signing keys should be published far enough in advance that they can be added
+//   by everyone before they are used to avoid service disruption.
+//
+// Resources:
+// - domains/apps/keyset
```

Comment:
Is it worth mentioning how Domains, Apps and KeySets are created? (IIRC, you have a pre-defined number of them in config files, right?)

Reply:
I'll add a line about it.  Thanks for thinking of it.